py/torch_tensorrt/dynamo/utils.py

```diff
@@ -661,3 +661,57 @@ def get_flat_args_with_check(
     flat_args_with_path, received_spec = pytree.tree_flatten_with_path((args, kwargs))
     flat_args = tuple(x[1] for x in flat_args_with_path)
     return flat_args, received_spec
+
+
+def get_output_meta_val(output: Any) -> List[Any]:
+    output_meta_val = []
+    if isinstance(output, torch.fx.node.Node):
+        if "val" in output.meta:
+            output_meta_val.append(output.meta["val"])
+    elif isinstance(output, (tuple, list)):
+        for node in output:
+            output_meta_val.extend(get_output_meta_val(node))
+    else:
+        raise ValueError(
+            f"expect torch.fx.node.Node or a tuple/list of torch.fx.node.Node type, got unexpected types: {type(output)=}"
+        )
+    return output_meta_val
+
+
+def set_output_meta_val(output: Any, outputs_meta_val: List[Any]) -> None:
+    if isinstance(output, torch.fx.node.Node):
+        assert len(outputs_meta_val) > 0
+        if "val" not in output.meta:
+            output.meta["val"] = outputs_meta_val[0]
+        outputs_meta_val.pop(0)
+    elif isinstance(output, (tuple, list)):
+        for node in output:
+            set_output_meta_val(node, outputs_meta_val)
+    else:
+        raise ValueError(
+            f"expect torch.fx.node.Node or a tuple/list of torch.fx.node.Node type, got unexpected types: {type(output)=}"
+        )
```

Comment:
We already have set_metadata and get_metadata functions in pass_utils.py. Consider merging them and using them across lowering pass for consistency